mautic/mautic
# Update iplookup to check do not sell list

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | Y
| New feature/enhancement? (use the a.x branch)      | N
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | Y 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
In this PR we are updating IpLookup to check Do Not Sell list.



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Add the path for the do not sell list to your config ('maxmind_do_not_sell_list_path' => 'some/file/path.json')
3. Run mautic:donotsell:download 
4. The Do Not sell list should have downloaded to the path you specified
5. Add an IP to your ip_address table that matches one that is in the DNS list and associate it with a contact. In the 'details' column, add some data for city, state, country and/or zip
6. Make sure the contact has no data in the city, state, country or zip code fields, (all should be NULL)
7. Make some change to the contact (not the address info) and save
8. On entity save, the Lead model will try to lookup data from the IP and add it to the contact. If everything works, then the city, state, country and zip fields should remain blank.
